### DISCUSSION TITLE
Display zero width characters in diffs when added or removed.

### DISCUSSION THREAD

**Discussion by Wopple** [community/community#10966] [Pull Requests]
I discovered some zero width characters in my file thanks to my editor, and so I removed them. While `git diff` is able to show those characters, the diff I see on github does not show those characters. Instead, it looks like an updated line with no apparent changes. In particular, the offending character was: `<U+FEFF>`.

**Comment by darcyclarke** [comment]
Notably, these characters probably need to be displayed in a more significant way outside of just diffs/PRs... we should be actively communicating to end-users when they're looking at _any_ code that contains these.

**Comment by rickycodes** [comment]
I really like the way [`bat`](https://github.com/sharkdp/bat) handles this stuff:

![image](https://user-images.githubusercontent.com/675259/159108525-6b7fba33-fad2-4774-80b5-51bbd10faa22.png)

**Reply by Marychoffel22** [reply]
Pull requests

**Comment by xyb** [comment]
I wholeheartedly agree (+1) that this feature would be extremely valuable. In fact, I've encountered another case in the cpython repository (https://github.com/python/cpython/pull/123933) where this capability would have been incredibly useful.
